### DISCUSSION TITLE
How can we distinguish beteen webpack.compiler and Rspack.compiler?

### DISCUSSION THREAD

**Discussion by GiveMe-A-Name** [web-infra-dev/rspack#2640] [Q&A] [answered]
As we known, the Rspack is progressively aligning webpack plugin api. 
Sometimes, We want to write a custom plugin that not only support rspack but also support Webpack.
In some situations, we custom logic for rspack & webpack, like this:
```js
apply(compiler) {
  if(isRspack) {
    // doSomethingWithRspack
  } else if(isWebpack) {
    // doSomethingWithWebpack
  }
}
```
So in the plugin we need distinguish the bundler.
We observed the `Rspack.compiler.webpack` has a `rspackVersion` attribute, but we think it's not a good way to distinguish beteen them. `Rspack.compiler.webpack.rspackVersion` maybe a unstable field.
So what is a good way to distinguish them?

**Comment by h-a-n-a** [comment, marked as answer]
`rspackVersion` is supposed to be a stable field with different version for each release. So I will recommend to use this field.
